src/core_modules/capture-core/components/D2Form/D2Form.component.js

```diff
@@ -135,21 +121,41 @@ class D2Form extends React.PureComponent<Props> {
     )
 
     render() {
-        const { formFoundation, id, classes, ...passOnProps } = this.props;
-        const formHorizontal = this.props.formHorizontal;
+        const {
+            formFoundation,
+            id,
+            classes,
+            isFormInReduxStore,
+            formHorizontal,
+            ...passOnProps
+        } = this.props;
         const metaDataSectionsAsArray = Array.from(formFoundation.sections.entries()).map(entry => entry[1]);
 
         const sections = metaDataSectionsAsArray.map(section => (formHorizontal ? this.renderHorizontal(section, passOnProps) : this.renderVertical(section, passOnProps, classes)));
 
         return (
             <React.Fragment>
-                {sections}
+                {
+                    isFormInReduxStore
+                        ?
+                        sections
+                        :
+                        log.error(
```

Comment:
log an error if the person who is using this component hasnt add the form id under the right reducer.